salt/grains/core.py

```diff
@@ -1156,6 +1156,23 @@ def os_data():
         grains.update(_windows_cpudata())
         grains.update(_windows_virtual(grains))
         grains.update(_ps(grains))
+
+        if 'Server' in grains['osrelease']:
```

Comment:
My concern here is not whether it works on Windows. My concern is whether or not this ends up modifying grains for non-Windows systems which have the string `Server` in their `osrelease` grain.